### ISSUE TITLE
Proposal: Move to atpm instead of using submodules

### ISSUE THREAD

**Issue by dunkelstern** [AnarchyTools/atbuild#91]
Missing Sub-modules would trigger a warning to run `atpm fetch` to get them and we would not have the `git submodule init && git submodule update` dance for all recursive dependencies. It would make clear that `atpkg` needs `atfoundation` and `atbuild` needs it too, currently `atfoundation` gets pulled in implicitly by `atpkg` which might confuse people.

Additionally we would not have `update xyz` commits everywhere. Just the lockfile would update (or if we track the master branch instead of a version, we could skip that too, but we would not get a reproducible build that way)

Bootstrapping would be no problem as `atpm` would only git clone the deps and symlink their `external` directory. That would be 2 lines in `bootstrap/build.sh` per dependency (so 4 lines currently)

**Comment by drewcrawford** [comment]
👍 , send PR

> Additionally we would not have update xyz commits everywhere. Just the lockfile would update 

I think this would still be the case.  At its core, a git submodule is just a lockfile itself, called `.gitmodules`.  So under atpm regime we continue to need update commits in all the cases we need them now I think?  But that's not a reason not to do it.

We should be dogfooding our own stuff I think, I have been avoiding that but mostly due to the static linking issue.  As that is resolved, that clears the only concern I have about doing this.

**Comment by dunkelstern** [comment]
Correct, but we only have one lock file that is updating all the time not the one in atbuild and another one in atpkg
